doc/release/upcoming_changes/26081.improvement.rst

```diff
@@ -0,0 +1,44 @@
+``ndarray`` Shapes are now covariant and bound
+---------------------------------------------------------------
+Static shape typing for ``ndarray`` is a long-term effort that continues 
+with this change.  Array shape can be described with a tuple of ints, but
+both ``TypeVar`` and ``TypeVarTuple`` provide some useful functionality.
+The former allows bounds and type variance, whereas the latter allows
+unpacking.  ``ndarray`` is built with a ``TypeVar``, and this change
+provides the appropriate bounds and variance.  It allows users
```

Comment:
`TypeVar` and `TypeVarTuple` are implementation details, as they won't be needed anymore once support for python versions below 3.12 has been dropped.
I believe that the implementation-agnistic terms *type parameter* and *variadic type parameters* are better suited here.

Reply:
Good point.  I was trying to include too much of the PR/development thought process in the release notes, but rereading this a week later it seems forced.  I'll take another go at it.